contracts/contracts/MACI.sol

```diff
@@ -133,14 +133,15 @@ contract MACI is IMACI, DomainObjs, Params, Utilities, Ownable {
     bytes memory _initialVoiceCreditProxyData
   ) public {
     // ensure we do not have more signups than what the circuits support
-    if (numSignUps == uint256(STATE_TREE_ARITY) ** uint256(stateTreeDepth)) revert TooManySignups();
+    if (numSignUps == uint256(TREE_ARITY) ** uint256(stateTreeDepth)) revert TooManySignups();
```

Comment:
Should we do `>=` here?

Reply:
Would you mind expanding why? 
I have been thinking `numSignups` nor `numMessages` can never be more than what we support in the circuits (`uint256(TREE_ARITY) ** uint256(stateTreeDepth)` and `maxValues.maxMessages`) so thought that `==`as condition would be enough, but maybe I'm missing something?

Reply:
I agree that it couldn't be more than the max values, and if they surpass the max value, they must be equal to the max values first (since they are added 1 once a time). But we also don't want it to become over the max value? To prevent maybe user accidentally modify it somewhere else to a very big value, that's why I suggest to change it to `>=`. I'm also curious why we only use `==`? Is it because of the gas fee? 
Maybe I'm thinking too much. If the condition I mentioned above will never happen, just ignore this recommendation.

Reply:
There are no instances where these variables are increased without first checking that we have not reached the limit, aside from the `init` function of Poll. In terms of gas costs these would increase slightly, but this is a very small value, don't think this was done because of gas costs (`eq` opcode vs `LT` and `ISZERO` so 3 gas vs 6 gas)

Having said that, I just saw that we have a problem [here](https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/MACI.sol#L136) we can have 1 signup more than what we support in the circuits, because in the constructor we add the [`BLANK_STATE_LEAF_HASH`](https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/MACI.sol#L104)

What you suggest here, we increase `numSignups` in constructor, or add a -1 to the check `numSignUps - 1 == uint256(STATE_TREE_ARITY) ** uint256(stateTreeDepth)`?

Reply:
Actually I just simply suggest to check `numSignUps >= uint256(TREE_ARITY) ** uint256(stateTreeDepth)`, just in case there's any condition that the numSignUps would be larger than the max value.

Reply:
Even if based on our current code it cannot happen? 

ps. I was also referring to our  MACI constructor where we enqueue one item inside the `stateAq` but do not increase `numSignups` - I have added a line to add +1 to `numSignups` after the first enqueue to avoid the possibility of `numSignups` being off by one

Reply:
In my opinion, yes, to avoid any accidence... but if you think this is unnecessary, we could keep it as it is now.

Reply:
I currently believe it is unnecessary, given I see no way to actually go > the limit, but if you see a way that this could go over the limit, then 100% we should add it, and prevent it from ever going over in the first place. What do you mean by accident here?

Reply:
I don't know, just in case? but if you think it's unnecessary, then just ignore it~~